src/AutoRest.CSharp/Common/Input/Source/ProtocolCompilationInput.cs

```diff
@@ -0,0 +1,99 @@
+        }
+
+        private ProtocolCompilationInput(Compilation compilation)
+            : base(compilation) { }
+
+        private protected override void FilterSymbols()
+        {
+            _methodSet = new List<IMethodSymbol>();
+            foreach (IModuleSymbol module in _compilation.Assembly.Modules)
+            {
+                foreach (var type in SourceInputHelper.GetSymbols(module.GlobalNamespace))
+                {
+                    if (type is INamedTypeSymbol typeSymbol && IsClient(typeSymbol))
+                    {
+                        foreach (var member in typeSymbol.GetMembers())
+                        {
+                            if (member is IMethodSymbol methodSymbol && IsProtocolMethod(methodSymbol))
+                            {
+                                _methodSet.Add(methodSymbol);
+                            }
+                        }
+                    }
+                }
+            }
+        }
+
+        internal override IMethodSymbol? FindMethod(string namespaceName, string typeName, string methodName, IEnumerable<CSharpType> parameters)
+        {
+            if (_methodSet == null)
+            {
+                FilterSymbols();
+            }
+
+            var methods = _methodSet.Where(m =>
+                m.ContainingNamespace.ToString() == namespaceName &&
+                m.ContainingType.Name == typeName &&
+                m.Name == methodName).ToArray();
+            if (methods.Length == 0)
+            {
+                return null;
+            }
+            else if (methods.Length == 1)
+            {
+                return methods.First();
+            }
+            else
+            {
+                foreach (var method in methods)
+                {
+                    var existingParameters = method.Parameters;
+                    var parametersCount = parameters.Count();
+                    if (existingParameters.Length - 1 != parametersCount)
+                    {
+                        continue;
+                    }
+
+                    int index = 0;
+                    foreach (var parameter in parameters)
+                    {
+                        if (SourceInputHelper.IsEqualType((INamedTypeSymbol)existingParameters[index].Type, parameter))
+                        {
+                            break;
+                        }
+                        ++index;
+                    }
+
+                    if (index == parametersCount)
+                    {
+                        return method;
+                    }
+                }
+                return null;
+            }
+        }
+
+        private bool IsClient(INamedTypeSymbol type) => type.Name.EndsWith("Client");
+        private bool IsProtocolMethod(IMethodSymbol method) => method.Parameters.Length > 0 && method.Parameters.Last().Name == "context";
```

Comment:
Maybe we should be precise and check if the last parameter is `RequestContext context`?

Reply:
The problem is: `RequestContext` is in `Azure.Core`, when we load it as a symbol, we don't know its namespace name.

Reply:
Currently there is no solution because we don't know the namespace of RequestContext

Reply:
I still did not get why we have a blocker on checking the type here.